### PR TITLE
Add "genjobs" CLI tool.

### PR DESCRIPTION
**Pull request by clarketm**
Generalized, extensible CLI tool to generate private security job(s) from a set of existing jobs.

### CODE REVIEW

File: prow/cmd/genjobs/genjobs.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+// validateJob validates that the job passes validation and should be converted.
+func validateJob(name string, patterns []string) bool {
+	if jobBlacklist.Has(name) || (len(jobWhitelist) > 0 && !jobWhitelist.Has(name)) || !isMatchBranch(patterns) {
+		return false
+	}
+
+	return true
+}
+
+// validateFlags validates that the command-line flags.
```
Comment (michelle192837) [inline prow/cmd/genjobs/genjobs.go]: nit: Cut 'that'?

File: prow/cmd/genjobs/genjobs.go [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+func validateFlags() {
+	var err error
+
+	if len(orgMap) == 0 {
+		printErrAndExit("-m, --map option is required.", 1)
```
Comment (michelle192837) [inline prow/cmd/genjobs/genjobs.go]: 'mapping'? (Or is the README incorrect?)

File: prow/cmd/genjobs/genjobs.go [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+	}
+
+}
+
+func printErrAndExit(msg string, code int) {
+	_, _ = fmt.Fprintln(os.Stderr, msg)
+	os.Exit(code)
+}
+
+// isMatchBranch validates that the branch for a job pass validation and should be converted.
```
Comment (michelle192837) [inline prow/cmd/genjobs/genjobs.go]: nit: passes?

File: prow/cmd/genjobs/genjobs.go [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+
+	}
+
+	return ""
+}
+
+// cleanOutPath deletes all files as the specified output path.
+func cleanOutPath(p string) {
+	for _, org := range orgMap {
+		_ = os.RemoveAll(filepath.Join(p, org))
```
Comment (michelle192837) [inline prow/cmd/genjobs/genjobs.go]: Shouldn't this output something if cleaning fails?

**Review by michelle192837 — APPROVED** [review]
/hold

File: prow/cmd/genjobs/genjobs.go (hunk 2 above)
Comment (clarketm) [reply prow/cmd/genjobs/genjobs.go]: Thanks, yes it should be `--mapping`. 🤦‍♂

File: prow/cmd/genjobs/genjobs.go (hunk 4 above)
Comment (clarketm) [reply prow/cmd/genjobs/genjobs.go]: If cleaning fails, conflicting files will still be overwritten, so it is not an error but I agree that it is a good idea to output a warning message to stderr to user.

Comment (clarketm) [reply prow/cmd/genjobs/genjobs.go]: Added printing to stderr in appropriate places 👍

**Review by michelle192837 — APPROVED** [review]
/hold cancel

File: prow/cmd/genjobs/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  -b, --repo-blacklist strings   Repositories to blacklist in generation process.
+  -w, --repo-whitelist strings   Repositories to whitelist in generation process.
+      --ssh-key-secret string    GKE cluster secrets containing the Github ssh private key. (default "ssh-key-secret")
+```
+
+## Example
+
+Translate all public jobs in `istio` organization to private jobs in `istio-private` organization in `./jobs` directory:
+
+```bash
```
Comment (fejta) [inline prow/cmd/genjobs/README.md:31]: console if you keep the `$`

File: prow/cmd/genjobs/README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+$ go run genjobs.go --mapping istio=istio-private --input ./jobs --output ./jobs
```
Comment (fejta) [inline prow/cmd/genjobs/README.md:32]: Remove the `$` if you stick with bash

Comment (clarketm) [reply prow/cmd/genjobs/README.md:32]: Thanks. Will make the update in a follow-up when I add more instructions and examples to the readme.

**Comment by clarketm** [comment]
fix #1876

**Pull request merged**